common/src/main/java/com/mx/path/core/common/remote_deposits/InsufficientFundsException.java

```diff
@@ -0,0 +1,26 @@
+package com.mx.path.core.common.remote_deposits;
+
+import com.mx.path.core.common.exception.PathRequestException;
+
+/**
+   * Exception thrown when an account has insufficient funds for a remote deposits.
+ */
+public class InsufficientFundsException extends PathRequestException {
+
+  public InsufficientFundsException() {
+    super();
+    initialize();
+  }
+
+  public InsufficientFundsException(String message, Throwable cause) {
+    super(message, cause);
+    initialize();
+  }
+
+  // Common initialization logic
+  private void initialize() {
+    setUserMessage(ErrorConstants.INSUFFICIENT_FUNDS_MESSAGE);
+    setCode(ErrorConstants.INSUFFICIENT_FUNDS_CODE);
+    withHeader("ErrorSubCode", ErrorConstants.INSUFFICIENT_FUNDS_CODE);
```

Comment:
This name suggests that there is an ErrorCode? Seems like these errors would be categorized by feature. Probably a good way to package them up as well.

Other header names...
Do we want to follow the `X-` custom header naming convention? Although, I don't think we've been doing that. :(

Other ideas:

`MX-Feature-Error-Code`

Reply:
We may also want to consider returning the feature as a response header. 🤔

Reply:
@mattnichols Thanks for your insights!
Regarding header naming conventions, I think sticking to our current practice with X- is fine for consistency. The MX-Feature-Error-Code header is a good idea, and I also like the suggestion of including the feature name as a response header.